### PR TITLE
add Docker to the project

### PR DESCRIPTION
**Pull request by nikitaabo**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,8 @@
+MYSQLDB_DATABASE=book_store
+MYSQLDB_USER=bestuser
+MYSQLDB_PASSWORD=bestuser
+MYSQLDB_ROOT_PASSWORD=rootpassword
+
+SPRING_DATASOURCE_URL=jdbc:mysql://mysql:3306/${MYSQLDB_DATABASE}
+SPRING_DATASOURCE_USERNAME=${MYSQLDB_USER}
+SPRING_DATASOURCE_PASSWORD=${MYSQLDB_PASSWORD}
```
Comment (okuzan) [inline .env]: never saw using env vars in env file :)
why don't you use them in docker compose?

File: .env [hunk 2]
```diff
@@ -0,0 +1,8 @@
+MYSQLDB_DATABASE=book_store
+MYSQLDB_USER=bestuser
+MYSQLDB_PASSWORD=bestuser
+MYSQLDB_ROOT_PASSWORD=rootpassword
+
+SPRING_DATASOURCE_URL=jdbc:mysql://mysql:3306/${MYSQLDB_DATABASE}
```
Comment (okuzan) [inline .env]: extract only port to env var, move other parts to docker compose

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  mysql:
+    image: mysql:8.0
+    restart: unless-stopped
+    environment:
+      MYSQL_DATABASE: ${MYSQLDB_DATABASE}
+      MYSQL_USER: ${MYSQLDB_USER}
+      MYSQL_PASSWORD: ${MYSQLDB_PASSWORD}
+      MYSQL_ROOT_PASSWORD: ${MYSQLDB_ROOT_PASSWORD}
+    ports:
+      - "3306:3306"
```
Comment (okuzan) [inline docker-compose.yml]: to env

**Review by okuzan — CHANGES_REQUESTED** [review]
git rm --cached .env
add to .gitignore
add .env.template with actual values removed, only var names
env contains secrets and shouldn't be in version control

File: .env (hunk 1 above)
Comment (nikitaabo) [reply .env]: 2 a.m.  + chat gpt = result like this )

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  mysql:
+    image: mysql:8.0
+    restart: unless-stopped
+    environment:
+      MYSQL_DATABASE: ${MYSQLDB_DATABASE}
+      MYSQL_USER: ${MYSQLDB_USER}
+      MYSQL_PASSWORD: ${MYSQLDB_PASSWORD}
+      MYSQL_ROOT_PASSWORD: ${MYSQLDB_ROOT_PASSWORD}
+    ports:
+      - "${MYSQLDB_PORT}:3306"
```
Comment (Elena-Bruyako) [inline docker-compose.yml]:
```suggestion
      - $MYSQLDB_LOCAL_PORT:$MYSQLDB_DOCKER_PORT
```

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    healthcheck:
+      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
+      interval: 30s
+      timeout: 30s
+      retries: 3
+
+  app:
+    build: .
+    ports:
+      - "8081:8080"
```
Comment (Elena-Bruyako) [inline docker-compose.yml]:
```suggestion
      - $SPRING_LOCAL_PORT:$SPRING_DOCKER_PORT
      - $DEBUG_PORT:$DEBUG_PORT
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by ahoienko — APPROVED** [review]
Good job!

**Pull request merged**